nat-lab/tests/telio.py

```diff
@@ -774,12 +773,17 @@ def get_stdout(self) -> str:
         assert self._process
         return self._process.get_stdout()
 
+    def get_features(self) -> TelioFeatures:
+        assert self._telio_features
+        return self._telio_features
+
     async def stop_device(self, timeout: float = 5) -> None:
         await asyncio.wait_for(self._write_command(["dev", "stop"]), timeout)
         self._interface_configured = False
-        assert Counter(self.get_runtime().get_started_tasks()) == Counter(
+        diff = set(self.get_runtime().get_started_tasks()) - set(
```

Comment:
Counter and set works differently here. Counter was chosen for a reason, to catch multiple occurrences of same task starting or stopping. For example, if 1 task starts 2 times, `set` will basically skip it, but `Counter` not.

```
from collections import Counter

start_task_list = ["task1", "task1", "task2"]
stop_task_list = ["task1", "task2"]

print(set(start_task_list) == set(stop_task_list))  # True
print(Counter(start_task_list) == Counter(stop_task_list))  # False

print(set(start_task_list) - set(stop_task_list))  # set()
print(Counter(start_task_list) - Counter(stop_task_list))  # Counter({'task1': 1})
```

Reply:
Would be nice to leave assert logic, so we could have `diff` printed in logs, but `Counter` should be brought back imo